zboxcore/sdk/writemarker_mutex.go

```diff
@@ -184,25 +184,26 @@ func (m *WriteMarkerMutex) lockOne(ctx context.Context, body io.Reader, url stri
 			TLSHandshakeTimeout: resty.DefaultDialTimeout,
 		})}
 
-	r := resty.New(func(req *http.Request, resp *http.Response, respBody []byte, cf context.CancelFunc, err error) error {
-		if err != nil {
-			return err
-		}
+	r := resty.New(options...)
 
-		if resp.StatusCode != http.StatusOK {
-			return errors.Throw(constants.ErrNotLockedWritMarker, fmt.Sprint(resp.StatusCode, ":", string(respBody)))
-		}
+	r.DoPost(ctx, body, url).
+		Then(func(req *http.Request, resp *http.Response, respBody []byte, cf context.CancelFunc, err error) error {
```

Comment:
I'm afraid the handler set through `.Then()` that is called after the `DoPost()` will not be triggered here. The `Then()` will only set the handle, but since the post process has already finished, thus the handler will not be called.

We can keep the handler as the first param of `resty.New()` just like we did before, sorry to repeat, but what I mean for 'initialize resty instance with default settings' was something like this:

```go
// in resty package
func New(handle Handle, opts ...Option) *Resty {
	r := &Resty{
		handle:  handle,
		timeout: DefaultRequestTimeout,
		retry:   DefaultRetry,
		transport: &http.Transport{
			Dial: (&net.Dialer{
				Timeout: DefaultDialTimeout,
			}).Dial,
			TLSHandshakeTimeout: DefaultDialTimeout,
		},
	}

	for _, option := range opts {
		option(r)
	}

	r.client = CreateClient(r.transport, r.timeout)

	return r
}

``` 

So the caller can use it simply as show below

```sh
r := resty.New(handler)

r.DoPost() // or something else
```

The options should only be used when we want different settings, such as different timeout or retry, while the `Transport` seems is always the same, and we can even put it as a global variable so we don't have to create it for every restry instance unless we want different settings.

Reply:
Then is triggered by Wait, not Do.Because handle should be thread -safe. That is why it is called one by one in wait channel instead of parallel request in Do So  chain-method works too, and is more grateful.

Reply:
DefaultValues are fixed.